File: linguagem_java/encapsulamento/esporte/Esporte.java
package encapsulamento.esporte;

public class Esporte {
    String nome;
    String modalidade;

    // Construtor da classe
    public Esporte(String nomeDoEsporte, String modalidadeEsportiva) {
        setNome(nomeDoEsporte);
        setModalidade(modalidadeEsportiva);
    }

    // Getter e setter de nome
    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    // Getter e setter de modalidade
    public String getModalidade() {
        return modalidade;
    }

    public void setModalidade(String modalidade) {
        this.modalidade = modalidade;
    }
}

File: linguagem_dart/encapsulamento/esporte/esporte.dart
class Esporte {
  String _nome = "";
  String _modalidade = "";

  // Construtor
  Esporte(String nomeDoPraticante, String modalidadeDesportiva){ 
    setNome = nomeDoPraticante;
    setModalidade = modalidadeDesportiva;
  }

  // Getter e setter de nome
  String get getNome => _nome;

  set setNome(String nomeDoEsporte) {
    _nome = nomeDoEsporte;
  }

  // Getter e setter de modalidade
  String get getModalidade => _modalidade;

  set setModalidade(String modalidadeDesportiva) {
    _modalidade = modalidadeDesportiva;
  }
}
